src/iframe/storage/unpartitioned-storage/local-storage.ts

```
@@ -42,4 +89,536 @@ export class LocalStorage extends EnhancedStorage {
+  /**
+   * Remove a cookie using CookieStore API or js-cookie fallback
+   */
+  private async removeCookie(key: string): Promise<void> {
+    try {
+      const cookieStore = this.getCookieStore();
+      if (cookieStore) {
+        await cookieStore.delete({
+          name: key,
+          path: this.COOKIE_STORE_OPTIONS.path,
+          partitioned: this.COOKIE_STORE_OPTIONS.partitioned,
+        });
+      } else {
+        Cookies.remove(key, this.COOKIE_OPTIONS);
+      }
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to remove cookie for key '${key}'`, e);
+      try {
+        Cookies.remove(key, this.COOKIE_OPTIONS);
+      } catch (fallbackError) {
+        if (LocalStorage.DEBUG_MODE) console.warn(`Fallback cookie remove also failed for key '${key}'`, fallbackError);
+      }
+    }
+  }
+
+  /**
+   * Remove all chunks for a given key
+   */
+  private async removeChunkedCookies(key: string): Promise<void> {
+    try {
+      const metaKey = `${key}${LocalStorage.CHUNK_META_SUFFIX}`;
+      const metaValue = await this.getCookieValue(metaKey);
+
+      if (metaValue) {
+        const chunkCount = parseInt(metaValue, 10);
+        if (!isNaN(chunkCount) && chunkCount > 0) {
+          // Remove all chunk cookies in parallel
+          const removePromises = [];
+          for (let i = 0; i < chunkCount; i++) {
+            removePromises.push(this.removeCookie(`${key}${LocalStorage.CHUNK_SUFFIX}${i}`));
+          }
+          await Promise.all(removePromises);
+        }
+
+        // Remove the metadata cookie
+        await this.removeCookie(metaKey);
+      }
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.error("Error in removeChunkedCookies", e);
+    }
+  }
+
+  /**
+   * Set an item in storage, with appropriate backup strategies
+   */
+  async setItem<T>(key: string, value: T, useRaw = false): Promise<void> {
+    if (!key) {
+      // TODO: Wouldn't this throw an error already? Otherwise, it should.
```

@pawanpaudel93 Any reason why you added a `return` instead of allowing the error to be thrown (I guess one will be thrown in some of the lines below if no key is passed).

---

Yes, I think this can be removed as I don't think I added this for some purpose.